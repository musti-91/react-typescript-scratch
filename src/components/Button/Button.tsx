import * as React from "react";

interface Props {
  title: string;
  onClick: () => void;
}

const Button = ({ onClick, title }: Props) => (
  <button onClick={() => onClick}>{title}</button>
);

export default Button;
