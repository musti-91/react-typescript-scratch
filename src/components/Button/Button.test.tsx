import * as React from "react";
import { shallow } from "enzyme";

import Button from "./Button";
describe("Button Component", () => {
  test("should match snapshot", () => {
    expect.assertions(1);

    const wrapper = shallow(<Button title={"MY BUTTON"} onClick={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
