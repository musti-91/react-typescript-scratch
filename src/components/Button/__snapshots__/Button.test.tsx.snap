// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<button
  onClick={[Function]}
>
  MY BUTTON
</button>
`;
